raiden/tests/utils/transfer.py

```diff
@@ -76,6 +77,36 @@ def get_channelstate(
     return channel_state
 
 
+def transfer_and_check(
```

Comment:
IMO this is not a good approach.

The function `tests.utils.transfer:transfer` has an argument `transfer_state`, which controls how the transfer will be left after the function returns. Currently, there are the following possible choices:

https://github.com/raiden-network/raiden/blob/c5e21c89c467e8713aeb5fd27e631398bbabd05e/raiden/tests/utils/transfer.py#L54-L63

One of which is `expired`, this means that however uses this functions has to know how they are implemented, and know that calling it with `transfer_state: TransferState = TransferState.Expired,` is invalid.

Also, note that among the available states `UNLOCKED = "unlocked"` is one of them, which expects everything to go as planned, and no failures to happen. There would be a good place to add `assert not has_unlock_failure` check.

With the above we will have less utilities, same functionality, less surprises on when the tool works.